### PR TITLE
Add builder log url for failed build

### PR DESCRIPTION
**Pull request by viveksyngh**
This commit adds builder pipeline log URL in github commit statues for
failed function build.

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Checklist:

I have:

- [ ] updated the documentation if required
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by viveksyngh** [comment]
This one I tested on my local cluster using this repo. https://github.com/viveksyngh/of-cloud-test-fn/commits/master

Check the GitHub commit status for the last commit.

File: github-status/handler.go [hunk 1]
```diff
@@ -128,6 +128,21 @@ func buildPublicStatusURL(status string, statusContext string, event *sdk.Event)
...
+
+		if len(gatewayPrettyURL) > 0 {
+			url = strings.Replace(gatewayPrettyURL, "user", "system", 1)
+			url = strings.Replace(url, "function", "pipeline", 1)
+
+		} else if len(publicURL) > 0 {
+			if strings.HasSuffix(publicURL, "/") == false {
+				publicURL = publicURL + "/"
+			}
+			url = publicURL + "/function/system-pipeline"
```
[the remark this answers is not in the document]
Comment (alexellis) [reply github-status/handler.go]: I think it's meant to be pipeline-log, it's the page which shows the specific build logs for this commit status.

**Comment by alexellis** [comment]
The name is `pipeline-log` rather than `pipeline`

**Comment by viveksyngh** [comment]
@alexellis I did not get you, should it be `system-pipeline-log` instead of `system-pipeline`?

**Comment by alexellis** [comment]
I thought so but I think I'm confusing the dashboard function with the logging function. Your changes should be fine.

File: github-status/handler.go [hunk 2]
```diff
@@ -128,6 +128,21 @@ func buildPublicStatusURL(status string, statusContext string, event *sdk.Event)
 				url = publicURL
 			}
 		}
+	} else if status == sdk.StatusFailure {
+		publicURL := os.Getenv("gateway_public_url")
+		gatewayPrettyURL := os.Getenv("gateway_pretty_url")
```
Comment (alexellis) [inline github-status/handler.go]: Quick question.. have you tested the code for `gateway_pretty_url`? I just need to know that works before merging/releasing.

Comment (alexellis) [reply github-status/handler.go]: Deploy the router component and then set an alias in your hosts file to make use of it. Then provide `gateway_pretty_url` in the YAML

Comment (viveksyngh) [reply github-status/handler.go]: I have not tested that particular feature, I will try that and test it.

Comment (viveksyngh) [reply github-status/handler.go]: tested for `pretty_url` as well, against same repo and it is working fine.

**Comment by alexellis** [comment]
@viveksyngh we've had some changes since you raised this PR. The logs should now go to the dashboard function.

This is the pretty URL format:

https://system.o6s.io/dashboard/viveksyngh/slack-webhook-fn/log?repoPath=viveksyngh/slack-webhook-fn&commitSHA=92a33883684cb5b6d4ecac5b77ab5f9089f23b39

And this is the regular format:

http://gateway/function/system-dashboard/viveksyngh/slack-webhook-fn/log?repoPath=viveksyngh/slack-webhook-fn&commitSHA=92a33883684cb5b6d4ecac5b77ab5f9089f23b39

@kenfdev can you confirm please?

Alex

**Comment by kenfdev** [comment]
Yes, ^those URLs should work 👍

**Comment by viveksyngh** [comment]
As I understood there are no functions called `system-pipeline` or `system-overview`, all of them are combined into one function `system-dashboard`.  Am I right?

**Comment by kenfdev** [comment]
@viveksyngh Almost correct. To be exact, the `system-dashboard` serves a JavaScript Single Page App + a simple proxy REST API which proxies requests to the gateway. The SPA requests the `system-dashboard` API and that request gets proxied to the `list-functions` and `pipeline-log` function on `of-cloud`.

`system-overview`, `system-list-functions`, `system-pipeline` functions defined [here](https://github.com/openfaas/openfaas-cloud/blob/master/dashboard/stack.yml) will eventually get deleted.

**Comment by viveksyngh** [comment]
So if the `pretty_url` is set, I should replace `user` with `system` and `function` with `dashboard` and then build the path after that? like `/<owner>/<functionName>/log?query_params`

**Comment by kenfdev** [comment]
Correct 👍

**Comment by viveksyngh** [comment]
I have updated it to follow new URL format.

**Comment by alexellis** [comment]
Thank you, does it work for both pretty URL and non-pretty URL now that I set up the DNS entries for you to test with on your team-serverless.xyz domain?

**Comment by alexellis** [comment]
Excellent contribution, thank you for taking the time to test this thoroughly.

**Pull request merged**